src/OrchardCore.Modules/OrchardCore.Media/Controllers/AdminController.cs

```diff
@@ -76,6 +77,19 @@ public async Task<ActionResult<IAsyncEnumerable<IFileStoreEntry>>> GetFolders(st
                 return NotFound();
             }
 
+            // create default folders if not exist
+
+            if (await _mediaFileStore.GetDirectoryInfoAsync(_mediaOptions.AssetsUsersFolder) == null)
+            {
+                await _mediaFileStore.TryCreateDirectoryAsync(_mediaOptions.AssetsUsersFolder);
+            }
+
+            if (await _authorizationService.AuthorizeAsync(User, Permissions.ManageOwnMedia)
+                && await _mediaFileStore.GetDirectoryInfoAsync($"{_mediaOptions.AssetsUsersFolder}/{User.Identity.Name}") == null)
```

Comment:
The UserId by default is probably better, because there may be many scenarios where the you don't want the user name exposed (I wouldn't want my name as part of a url, for example).

However, other sites may wish to, or use a custom claim for the name, so I would suggest we could add a service, `IUserAssetFolderNameProvider` which takes a `ClaimsPrincipal`, and by default returns the `NameIdentifier` value.

In this way someone can replace the service if they wish to use a different naming convention.

Reply:
IUserAssetFolderNameProvider added and by default return NameIdentifier.